frontend/src/metabase/redux/embed/embed.unit.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { type Dispatch, configureStore } from "@reduxjs/toolkit";
+
+import {
+  DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+  embed as embedReducer,
+  setInitialUrlOptions,
+} from "./embed";
+
+describe("embed reducer", () => {
+  describe("setInitialUrlOptions", () => {
+    it("should set default options", () => {
+      const store = createMockStore();
+
+      store.dispatch(setInitialUrlOptions({ search: "" }));
+
+      expect(store.getState().embed.options).toEqual(
+        DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
+      );
+    });
+
+    it("should set options from search", () => {
+      const store = createMockStore();
+
+      store.dispatch(
+        setInitialUrlOptions({ search: "top_nav=false&new_button=true" }),
+      );
+
+      expect(store.getState().embed.options.top_nav).toBe(false);
+      expect(store.getState().embed.options.new_button).toBe(true);
+    });
+
+    it("should ignore invalid options", () => {
+      const store = createMockStore();
+
+      store.dispatch(
+        setInitialUrlOptions({ search: "top_nav=false&invalid_option=123" }),
+      );
+
+      expect(store.getState().embed.options).not.toHaveProperty(
+        "invalid_option",
+      );
+    });
+
+    describe("entity_types", () => {
+      it('should accept multiple "entity_types" options', () => {
+        const store = createMockStore();
+
+        store.dispatch(
+          setInitialUrlOptions({
+            search: "entity_types=table&entity_types=model",
+          }),
+        );
+
+        // The default value is `["model", "table"]`, so we know these 2 types are set correctly.
+        expect(store.getState().embed.options.entity_types).toEqual([
+          "table",
+          "model",
```

Comment:
if you still haven't noticed, the values are swapped from the default value (`table` first vs `model` first)